contrib/mesos/pkg/executor/executor.go

```diff
@@ -212,10 +199,15 @@ func (k *KubernetesExecutor) Registered(driver bindings.ExecutorDriver,
 	}
 
 	if executorInfo != nil && executorInfo.Data != nil {
-		k.staticPodsConfig = executorInfo.Data
+		k.initializeStaticPodsSource(executorInfo.Data)
 	}
 
-	k.initialRegistration.Do(k.onInitialRegistration)
+	k.updateChan <- kubelet.PodUpdate{
```

Comment:
writing and closing at the same time to a chan will cause a data race

2015-09-22 17:30 GMT-04:00 Dr. Stefan Schimanski <notifications@github.com>:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/13036#discussion_r40146867>
> :
>
> >  	}
> >
> > -	k.initialRegistration.Do(k.onInitialRegistration)
> > +	k.updateChan <- kubelet.PodUpdate{
>
> With the little exception of writing to a closed channel which panics.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13036/files#r40146867>.
>